crates/task-impls/src/vid.rs

```diff
@@ -220,12 +217,24 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
             }
 
             HotShotEvent::TransactionsSequenced(encoded_transactions, metadata, view_number) => {
+                // get quorum committee for dispersal
+                let num_quorum_committee = self.membership.get_committee(view_number).len();
+
                 // TODO <https://github.com/EspressoSystems/HotShot/issues/1686>
-                let srs = test_srs(NUM_STORAGE_NODES);
-                // TODO We are using constant numbers for now, but they will change as the quorum size
-                // changes.
-                // TODO <https://github.com/EspressoSystems/HotShot/issues/1693>
-                let vid = VidScheme::new(NUM_CHUNKS, NUM_STORAGE_NODES, &srs).unwrap();
+                let srs = test_srs(num_quorum_committee);
+
+                // calculate the last power of two
+                // TODO change after https://github.com/EspressoSystems/jellyfish/issues/339
+                let chunk_size = {
+                    let mut power = 1;
+                    while (power << 1) <= num_quorum_committee {
+                        power <<= 1;
+                    }
+                    power
```

Comment:
Would it be more efficient to calculate the ceiling of `log2(num_quorum_committee)` than to use this while loop?

Reply:
https://doc.rust-lang.org/std/primitive.u64.html#method.next_power_of_two

Reply:
^ This is better. 👍

Reply:
I think we want the previous power of two, as the chunk size has to be less than the number of storage nodes or else disperse will panic.

But I will look into improving this

Reply:
Algorithm has been updated... hopefully it's a better solution